lib/ex_aws/request.ex

```diff
@@ -84,9 +86,11 @@ defmodule ExAws.Request do
     {:attempt, attempt + 1}
   end
 
-  # TODO: make exponential
-  # TODO: add jitter
   def backoff(attempt) do
-    :timer.sleep(attempt * 1000)
+    (@base_backoff_in_ms * :math.pow(2, attempt))
+    |> min(@max_backoff_in_ms)
+    |> trunc
+    |> :rand.uniform
```

Comment:
So, I think the issue here is that if you do `:rand.uniform(10000)` it's going to be _any_ number between 0 and 10000. We definitely want jitter, but jitter is supposed to be some level of fuzz around an otherwise increasing value.

We need to do some thing like:

```
base_value =
  (@base_backoff_in_ms * :math.pow(2, attempt))
  |> min(@max_backoff_in_ms)
  |> trunc

:timer.sleep(base_value + trunc(base_value / 10))
```

This gives us a 10% random jitter on each sleep value, while still guaranteeing a general curve.